python/src/aac/lang/lsp/server.py

```diff
@@ -0,0 +1,84 @@
+"""The Architecture-as-Code Language Server."""
+
+import logging
+from typing import Optional
+from pygls.server import LanguageServer
+import pygls.lsp.methods as methods
+
+from pygls.lsp import (
+    CompletionOptions,
+    CompletionParams,
+    Hover,
+    HoverParams,
+    InitializeParams,
+    InitializeResult,
+    ServerCapabilities,
+)
+
+from aac.lang.active_context_lifecycle_manager import get_active_context
+from aac.lang.language_context import LanguageContext
+from aac.lang.lsp.code_completion_provider import CodeCompletionProvider
+
+
+LOGGER: Optional[logging.Logger] = None
+SERVER: Optional[LanguageServer] = None
+ACTIVE_CONTEXT: Optional[LanguageContext] = None
+COMPLETION_PROVIDER: Optional[CodeCompletionProvider] = None
```

Comment:
I'm getting concerned with our use of global variables (even though they're only technically module-level globals) in various places. I think we need to start reconsidering how to approach this without using as many globals - perhaps we should start favoring passing objects as parameters when they're needed would be a good option instead of creating more globals?

Reply:
Will address in #227